test/flink_onprem_test.go

```diff
@@ -75,6 +75,87 @@ func (s *CLITestSuite) TestFlinkEnvironmentDelete() {
+		{args: "flink savepoint create savepoint1 --environment default", fixture: "flink/savepoint/create-savepoint-fail-none.golden", exitCode: 1},
+		{args: "flink savepoint create savepoint1 --application application1 --statement statement1", fixture: "flink/savepoint/create-savepoint-fail-no-env.golden", exitCode: 1},
+	}
+
+	runIntegrationTestsWithMultipleAuth(s, tests)
+}
+
+func (s *CLITestSuite) TestFlinkSavepointListOnPrem() {
+	tests := []CLITest{
+		// success scenarios
+		{args: "flink savepoint list --environment default --application application1", fixture: "flink/savepoint/list-successful.golden"},
+		{args: "flink savepoint list --environment default --statement statement1", fixture: "flink/savepoint/list-successful-statement.golden"},
+		// failure scenarios
+		{args: "flink savepoint list --environment default --statement statement1 --application application1", fixture: "flink/savepoint/list-fail-both.golden", exitCode: 1},
+	}
+
+	runIntegrationTestsWithMultipleAuth(s, tests)
+}
+
+func (s *CLITestSuite) TestFlinkSavepointDescribe() {
+	tests := []CLITest{
+		{args: "flink savepoint describe savepoint1 --environment default --application application1", fixture: "flink/savepoint/describe-success.golden"},
+		{args: "flink savepoint describe savepoint1 --environment default --statement statement1", fixture: "flink/savepoint/describe-success-statement.golden"},
+		{args: "flink savepoint describe invalid-savepoint --environment default --statement statement1", fixture: "flink/savepoint/describe-fail.golden", exitCode: 1},
+	}
+
+	runIntegrationTestsWithMultipleAuth(s, tests)
+}
+
+func (s *CLITestSuite) TestFlinkSavepointDeleteOnPrem() {
+	tests := []CLITest{
+		{args: "flink savepoint delete savepoint1 --environment default --application application1", input: "y\n", fixture: "flink/savepoint/delete-success.golden"},
+		{args: "flink savepoint delete savepoint1 --environment default --statement statement1", input: "y\n", fixture: "flink/savepoint/delete-statement-success.golden"},
+		{args: "flink savepoint delete savepoint1 --environment default --statement statement1 --force", fixture: "flink/savepoint/delete-force-success.golden"},
+	}
+
+	runIntegrationTestsWithMultipleAuth(s, tests)
+}
+
+func (s *CLITestSuite) TestFlinkDetachedSavepointCreate() {
+	tests := []CLITest{
+		{args: "flink detached-savepoint create savepoint1 --path abc/def", fixture: "flink/detached-savepoint/create-savepoint.golden"},
+		{args: "flink detached-savepoint create savepoint1", fixture: "flink/detached-savepoint/create-savepoint-nopath.golden", exitCode: 1},
+	}
+
+	runIntegrationTestsWithMultipleAuth(s, tests)
+}
+
+func (s *CLITestSuite) TestFlinkDetachedSavepointListOnPrem() {
+	tests := []CLITest{
+		{args: "flink detached-savepoint list", fixture: "flink/detached-savepoint/list-successful.golden"},
```

Comment:
Adding a serialized output test here will add coverage for `convertSdkDetachedSavepointToLocalSavepoint`.